### PR TITLE
Makefile.include: Add -Werror by default

### PR DESCRIPTION
**Pull request by gebart**
Typo caused Werror to be enabled for all builds _except_ buildtest.

Follow up question: Should we remove the `if` and use `-Werror` always?

### CODE REVIEW

**Comment by BytesGalore** [comment]
+1 for always `-Werror`

**Comment by OlegHahm** [comment]
Ah, and I wondered why changing something in `Makefile.buildtests` influenced building other apps...

:+1: for enabling it for everything.

**Comment by gebart** [comment]
@Kijewski you were thinking about users who want to write bad code should be allowed to, do you have any opinions on this?
I could be possible with some Makefile magic to only add `-Werror` on RIOT files, and let all user projects have whatever ugly code they want. I would however like to leave such a hack for a later PR.

**Comment by Kijewski** [comment]
Well, the easiest solution might be a new line in our example Makefiles
```make
# Fail on warnings. Can be overridden by `make WERROR=0`.
WERROR ?= 1
```
mimicking the `QUIET` variable.

Requires less magic and the user has a nice default if they copy the hello-world example.

**Comment by OlegHahm** [comment]
@gebart, are you planning to adapt?

**Comment by gebart** [comment]
On Jul 28, 2015 6:11 PM, "Oleg Hahm" <notifications@github.com> wrote:
>
> @gebart, are you planning to adapt?

I can adapt this next week. I'm on holidays until Sunday.

**Comment by OlegHahm** [comment]
Cool, no need to rush. Enjoy your holidays!

**Comment by hexluthor** [comment]
This doesn't fix the problem because the `buildtest` target is not present in the recursive call to `make`. Before, `-Werror` was always enabled. Now it's never enabled. See #3537.

**Comment by gebart** [comment]
@hexluthor You are correct. Closing in favour of #3537

**Comment by gebart** [comment]
Reopened.
Rebased and implemented the suggested `WERROR?=1` flag by @Kijewski.

**Comment by gebart** [comment]
crap

**Comment by gebart** [comment]
Reopened properly.

**Comment by gebart** [comment]
@hexluthor what do you think about the new proposal?

**Comment by hexluthor** [comment]
Looks good to me. :+1:

**Comment by OlegHahm** [comment]
Shouldn't we just make `WERROR = 1` the default and save the line in the example (and dist) Makefiles?

**Comment by gebart** [comment]
Moved WERROR to Makefile.include.

**Comment by gebart** [comment]
Added passing of WERROR variable to buildtest sub-make

File: Makefile.include [hunk 1]
```diff
@@ -64,10 +64,17 @@ else
...
+  AD=@
+  MAKEFLAGS += --no-print-directory
 else
-	AD=
+  AD=
+endif
+
+# Fail on warnings. Can be overridden by `make WERROR=0`.
+WERROR ?= 1
+export WERROR
```
Comment (OlegHahm) [inline Makefile.include]: Why is this export necessary?

Comment (gebart) [inline Makefile.include]: `make buildtest`'s call to sub-make didn't see the correct value for the `WERROR` variable without it (I tried it both as a bash variable `$${WERROR}` and as a make variable `$(WERROR)`)

**Comment by OlegHahm** [comment]
ACK

**Comment by gebart** [comment]
Squashed

**Pull request merged**